frontend/src/hooks/useSmartTrainerInterface.ts

```diff
@@ -49,61 +140,158 @@ export const useSmartTrainerInterface = (): SmartTrainer => {
   };
 
   const requestPermission = async () => {
-    const device = await navigator.bluetooth.requestDevice({
-      filters: [{ services: ['fitness_machine'] }], // Fitness machine 0x1826
-    });
+    dispatch({ type: 'set_connecting' });
 
-    setDevice(device);
+    const device = await navigator.bluetooth
+      .requestDevice({
+        filters: [{ services: ['cycling_power'] }],
+        optionalServices: ['fitness_machine'],
```

Comment:
This is the main thing - we will now primarily look for devices that announce the `cycling_power`service, but we still prefer the `fitness_machine` service. When we connect, we will first try to connect to the `fitness_machine` service, and if that fails, we will try to connect to the `cycling_power` service